### PR TITLE
.NET Spark - Getting Started Doc

### PR DESCRIPTION
**Pull request by bamurtaugh**
## Summary

Revamped the getting started doc for .NET for Apache Spark.
-Changed order of steps to flow more smoothly
-Added greater detail and images as examples
-Update some versioning and commands

-Overall changes to match content/flow found on .NET site: https://dotnet.microsoft.com/learn/data/spark-tutorial/intro

@mamccrea

### CODE REVIEW

File: docs/spark/tutorials/get-started.md [hunk 1]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
 > * Prepare your Windows environment for .NET for Apache Spark
-> * Download the **Microsoft.Spark.Worker**
-> * Build and run a simple .NET for Apache Spark application
+> * Write your first .NET for Apache Spark application
+> * Build and run your simple .NET for Apache Spark application
 
 ## Prepare your environment
 
-Before you begin, make sure you can run `dotnet`, `java`, `mvn`, `spark-shell` from your command line. If your environment is already prepared, you can skip to the next section. If you cannot run any or all of the commands, follow the steps below.
+Before you begin writing your app, you need to setup some prerequisite dependencies. If you can run `dotnet`, `java`, `mvn`, `spark-shell` from your command line environment, then your environment is already prepared and you can skip to the next section. If you cannot run any or all of the commands, follow the steps below.
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/b/below

> Don't use as an adjective preceding a noun (the below section) or following a noun (the code below). Use a hyperlink, or use later or the following.

```suggestion
Before you begin writing your app, you need to setup some prerequisite dependencies. If you can run `dotnet`, `java`, `mvn`, `spark-shell` from your command line environment, then your environment is already prepared and you can skip to the next section. If you cannot run any or all of the commands, follow the following steps.
```

File: docs/spark/tutorials/get-started.md [hunk 2]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
 ## Write a .NET for Apache Spark app
 
-1. Open **Visual Studio** and navigate to **File > Create New Project > Console App (.NET Core)**. Name the application **HelloSpark**.
+### 1. Create a console app
 
-2. Install the [Microsoft.Spark NuGet package](https://www.nuget.org/profiles/spark). For more information on installing NuGet packages, see [Different ways to install a NuGet Package](https://docs.microsoft.com/nuget/consume-packages/ways-to-install-a-package).
+In your command prompt, run the following commands to create a new console application:
 
-3. In **Solution Explorer**, open **Program.cs** and write the following C# code:
+```
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]: An identifier should be added here, either dotnetcli or console.

File: docs/spark/tutorials/get-started.md [hunk 3]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+### 2. Install NuGet package
+
+To use .NET for Apache Spark in an app, you need to install the Microsoft.Spark package. In your command prompt, run the following command:
 
-4. Build the solution.
+`dotnet add package Microsoft.Spark --version 0.6.0`
+
+### 3. Code your app
+
+Open `Program.cs` in Visual Studio or any text editor and replace all of the code with the following:
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
Open *Program.cs* in Visual Studio or any text editor and replace all of the code with the following:
```

File: docs/spark/tutorials/get-started.md [hunk 4]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+```csharp
+using Microsoft.Spark.Sql;
+
+namespace MySparkApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // Create a Spark session
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
            // Create a Spark session.
```

File: docs/spark/tutorials/get-started.md [hunk 5]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+            var spark = SparkSession
+                .Builder()
+                .AppName("word_count_sample")
+                .GetOrCreate();
+
+            // Create initial DataFrame
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
            // Create initial DataFrame.
```

File: docs/spark/tutorials/get-started.md [hunk 6]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+            DataFrame dataFrame = spark.Read().Text("input.txt");
+
+            // Count words
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
            // Count words.
```

File: docs/spark/tutorials/get-started.md [hunk 7]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+            var words = dataFrame
+                .Select(Functions.Split(Functions.Col("value"), " ").Alias("words"))
+                .Select(Functions.Explode(Functions.Col("words"))
+                .Alias("word"))
+                .GroupBy("word")
+                .Count()
+                .OrderBy(Functions.Col("count").Desc());
+
+            // Show results
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
            // Show results.
```

File: docs/spark/tutorials/get-started.md [hunk 8]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+            words.Show();
+
+            // Stop Spark session
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
            // Stop Spark session.
```

File: docs/spark/tutorials/get-started.md [hunk 9]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+            spark.Stop();
+        }
+    }
+}
+```
+
+### 4. Add data file
+
+Your app will be processing a file containing lines of text. Create an `input.txt` file in your `mySparkApp` directory, containing the following text:
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]:
```suggestion
Your app will be processing a file containing lines of text. Create an *input.txt* file in your *mySparkApp* directory, containing the following text:
```

File: docs/spark/tutorials/get-started.md [hunk 10]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
+
+```
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]: Not sure if `text` identifier should be added or keep this without an identifier.

File: docs/spark/tutorials/get-started.md [hunk 11]
```diff
@@ -16,104 +16,185 @@ In this tutorial, you learn how to:
...
 
 ## Run your .NET for Apache Spark app
 
-1. Open **PowerShell** and change the directory to the folder where your app is stored.
+1. Run the following command to build your application:
 
    ```powershell
-   cd <your-app-output-directory>
+   dotnet build
    ```
```
Comment (Youssef1313) [inline docs/spark/tutorials/get-started.md]: Make this dotnetcli instead of powershell ?

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: docs/spark/tutorials/get-started.md (hunk 2 above)
Comment (bamurtaugh) [reply docs/spark/tutorials/get-started.md]: Added console identifier.

File: docs/spark/tutorials/get-started.md (hunk 10 above)
Comment (bamurtaugh) [reply docs/spark/tutorials/get-started.md]: I can add the identifier if you think that'd make more sense, I'm open to suggestions.

File: docs/spark/tutorials/get-started.md (hunk 11 above)
Comment (bamurtaugh) [reply docs/spark/tutorials/get-started.md]: Done.

File: docs/spark/tutorials/get-started.md (hunk 10 above)
Comment (bamurtaugh) [reply docs/spark/tutorials/get-started.md]: I feel like adding the identifier makes sense/makes it more consistent across the rest of the doc, so I added it.

**Review by mamccrea — APPROVED** [review]

**Pull request merged**